src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry-preview/telemetry-preview.component.html

```diff
@@ -0,0 +1,101 @@
+<cd-loading-panel *ngIf="loading && !error"
+                  i18n>Loading configuration...</cd-loading-panel>
+<cd-alert-panel type="error"
+                *ngIf="loading && error"
+                i18n>The configuration could not be loaded.</cd-alert-panel>
+
+<div class="cd-col-form"
+     *ngIf="!loading && !error">
+  <form name="mgrModuleForm"
+        #frm="ngForm"
+        [formGroup]="telemetryPreviewForm"
+        novalidate>
+    <div class="card">
+      <div class="card-header"
+           i18n>Step 2 of 2: Telemetry report preview</div>
+      <div class="card-body">
+        <!-- Telemetry report -->
+        <div class="form-group row">
```

Comment:
All this part is common to the 'Step 1 of 2' of this wizard (and both component.ts files share lots of common code). Wouldn't it make sense to keep it everything into a single component? I mean, the multi-page is a layout concern (presentation), nothing to do with the underlying logic (configuring Telemetry module).

I checked how [Angular Material does this](https://material.angular.io/components/stepper/overview) and according to that it would look like this:
```html
<!-- being this example just 2 steps, an  *ngIf would be enough, but this is just as an example -->
<ng-container [ngSwitch]="step">
  <some-element *ngSwitchCase="1">
    ...
  </some-element>
  <some-element *ngSwitchCase="2">
    ...
  </some-element>
</ng-container>
```

Reply:
Thanks for the suggestion. It's fine with me - I'll take a look at it.